### PR TITLE
chore(deps): Update to use EdgeX 3.1.0 dependencies

### PR DESCRIPTION
**Pull request by lenny-intel**
## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md
- [ ] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## Issue this PR will close

close: #**issue_number**

### CODE REVIEW

File: loss-detection-app/res/device-mqtt/profiles/cv-roi.yaml [hunk 1]
```diff
@@ -15,5 +15,5 @@ deviceResources:
   - name: "cv-roi-event"
     description: "A CV ROI event"
     properties:
-      valueType: "string"
```
Comment (lenny-intel) [inline loss-detection-app/res/device-mqtt/profiles/cv-roi.yaml:18]: Core is expecting `object`, not `string` type

File: rtsf-at-checkout-event-reconciler/events/models.go [hunk 2]
```diff
@@ -136,7 +136,7 @@ func (eventsProcessing *EventsProcessor) GetCurrentStateMessage() []byte {
 	return eventsProcessing.currentStateMessage
 }
 
-//these custom Marshal functions are needed as json.Marshal() results in a stack overflow error from an infinite loop, due to cross-referencing of Associated RTTL/Scale Entries
+// these custom Marshal functions are needed as json.Marshal() results in a stack overflow error from an infinite loop, due to cross-referencing of Associated RTTL/Scale Entries
```
Comment (lenny-intel) [inline rtsf-at-checkout-event-reconciler/events/models.go:139]: Go 1.21 gofmt slightly different

File: rtsf-at-checkout-event-reconciler/main.go [hunk 3]
```diff
@@ -59,9 +60,7 @@ func (app *EventReconcilerAppService) CreateAndRunAppService(serviceKey string)
 	eventsProcessor.ResetEventsOccurrence()
 	eventsProcessor.InitWebSocketConnection(app.service, app.lc)
 
-	deviceNamesList := strings.TrimSpace(app.serviceConfig.Reconciler.DeviceNames)
-
-	deviceNames := strings.Split(deviceNamesList, ",")
+	deviceNames := util.DeleteEmptyAndTrim(strings.FieldsFunc(app.serviceConfig.Reconciler.DeviceNames, util.SplitComma))
```
Comment (lenny-intel) [inline rtsf-at-checkout-event-reconciler/main.go:63]: Fix code that was parsing the device names, which left a blank at the start of each name so the filter wasn't accepting any of the data from the devices.

File: rtsf-at-checkout-event-reconciler/res/configuration.yaml [hunk 4]
```diff
@@ -16,10 +16,10 @@ MessageBus:
...
 Reconciler:
   DeviceNames: "pos-rest, pos-mqtt, device-scale, scale-rest, scale-mqtt, cv-roi-rest, cv-roi-mqtt, rfid-roi-rest, rfid-roi-mqtt"
-  DevicePos: device-pos
-  DeviceScale: device-scale
-  DeviceCV: device-cv-roi
-  DeviceRFID: device-rfid-roi
+  DevicePos: pos
+  DeviceScale: scale
+  DeviceCV: cv-roi
+  DeviceRFID: rfid-roi
```
Comment (lenny-intel) [inline rtsf-at-checkout-event-reconciler/res/configuration.yaml:22]: Match names in the device files.

File: rtsf-at-checkout-loss-detector/functions/notify.go [hunk 5]
```diff
@@ -53,7 +53,7 @@ func NotifySuspectList(ctx interfaces.AppFunctionContext, data interface{}) (boo
 		},
 		securityCategory,
 		content,
-		"Loss Detector",
+		"Loss-Detector",
```
Comment (lenny-intel) [inline rtsf-at-checkout-loss-detector/functions/notify.go:56]: EdgeX limits character set for the name which doesn't include space.

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Review by brian-intel — APPROVED** [review]

**Pull request merged**